### PR TITLE
Cleanup the Job controller handling of terminating pods

### PR DESCRIPTION
**Pull request by mimowo**
/sig apps

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

To make the code easier to understand and modify. The simplifications will be useful to implement https://github.com/kubernetes/enhancements/pull/3967 (and potentially https://github.com/kubernetes/enhancements/pull/3940).

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

This PR deletes one copy of the TODO(#113855). The other remains inside the `isPodFailed` function. 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/controller/job/job_controller.go [hunk 1]
```diff
@@ -1021,7 +1006,7 @@ func (jm *Controller) trackJobStatusAndRemoveFinalizers(ctx context.Context, job
 				needsFlush = true
 				uncountedStatus.Succeeded = append(uncountedStatus.Succeeded, pod.UID)
 			}
-		} else if pod.Status.Phase == v1.PodFailed || considerTerminated {
+		} else if considerPodFailed || finishedCond != nil {
 			ix := getCompletionIndex(pod.Annotations)
```
Comment (alculquicondor) [inline pkg/controller/job/job_controller.go:1010]: Add a comment: When the job is considered finished, every non-terminated pod is considered failed.

**Review by alculquicondor — COMMENTED** [review]
/approve

File: pkg/controller/job/job_controller.go (hunk 1 above)
Comment (mimowo) [reply pkg/controller/job/job_controller.go:1011]: done

**Pull request merged**